### PR TITLE
Fwmt 559 cancel request

### PR DESCRIPTION
**Pull request by ChrisHardmanONS**
combined Jacobs and my tests in one branch

### CODE REVIEW

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                    + "\t\t<reason>deleted for test</reason>\n"
+                    + "\t</actionCancel>\n"
+                    + "</ins:actionInstruction>";
+
+
+    private String username = "guest";
+    private String password = "guest";
+    private String hostname = "localhost";
+    private int rmPort = 6672;
+    private String virtualHost = "/";
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java]: put all these in a yml

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+import static org.junit.Assert.assertEquals;
+
+public class OHSSteps {
+
+    private final String EXCHANGE_NAME = "action-outbound-exchange";
+    private final String ROUTING_KEY = "Action.Field.binding";
+    private final String XMLCREATE = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n"
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java]: put in a file and read it

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            + "        <events>\n"
+            + "            <event>CASE_CREATED : null : SYSTEM : Case created when Initial creation of case</event>\n"
+            + "        </events>\n"
+            + "        <exerciseRef>202103</exerciseRef>\n"
+            + "        <userDescription>March 2021</userDescription>\n"
+            + "        <surveyName>Monthly Wages and Salaries Survey</surveyName>\n"
+            + "        <surveyRef>LMS</surveyRef>\n"
+            + "    </actionRequest>\n"
+            + "</ns2:actionInstruction>";
+    private final String XMLCANCEL =
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java]: put in a file

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    private String username = "guest";
+    private String password = "guest";
+    private String hostname = "localhost";
+    private int rmPort = 6672;
+    private String virtualHost = "/";
+
+
+    @Before
+    public void rest_mock() throws IOException {
+        URL url = new URL("http://localhost:9099/logger/reset");
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java]: yml

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+        conn.setRequestMethod("GET");
+        if (conn.getResponseCode() != 200) {
+            throw new RuntimeException("Failed : HTTP error code : "
+                    + conn.getResponseCode());
+        }
+    }
+
+    @Given("^RM sends LMS (\\d+) \"([^\"]*)\" case samples to the Gateway$")
+    public void rm_sends_LMS_case_samples_to_the_Gateway(int arg1, String type)
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java]: rename LMS to OHS

Comment (spdiaz1000) [reply src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java]: what is this arg1 u speak of

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+        Connection connection = factory.newConnection();
+        Channel channel = connection.createChannel();
+
+        if (type.equals("create")) {
+            message = XMLCREATE;
+        } else if (type.equals("cancel")) {
+            message = XMLCANCEL;
+        } else {
+            System.out.println(type+" is an unknown message please use create or cancel");
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java]: logger please

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+            }
+        }
+
+        channel.close();
+        connection.close();
+    }
+
+    @Then("^loaded in TM (\\d+)$")
+    public void loaded_in_TM(int jobs) throws IOException {
+        URL url = new URL("http://localhost:9099/logger/allMessages");
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java]: yml the url

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+                (conn.getInputStream())));
+        StringBuilder sb = new StringBuilder();
+        String line;
+
+        try  {
+            while ((line = bufferedReader.readLine()) != null){
+                sb.append(line + "\n");
+            }
+        }catch (IOException e){
+            e.printStackTrace();
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java]: throw this exception, let if fail

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+        }
+
+        String result = sb.toString();
+
+        ObjectMapper mapper = new ObjectMapper();
+        List<MockMessage> message = mapper.readValue(result, new TypeReference<List<MockMessage>>(){});
+
+
+        assertEquals(message.size(),jobs);
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/steps/OHSSteps.java]: expected first then actual

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/Constatnts.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
+package uk.gov.ons.fwmt.acceptancetests.utils;
+
+import lombok.Data;
+
+@Data
+public class Constatnts {
+
+  public static final String LMS_RESPONSE = "<?xml version=\"1.0\" encoding=\"utf-8\"?><soap:Envelope xmlns:soap=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"><soap:Body><request xmlns=\"http://schemas.consiliumtechnologies.com/services/mobile/2009/03/messaging\"><Id>4ae43740-c9c8-4258-8052-49fcc4d5fb39</Id><Content>&lt;![CDATA[&lt;?xml version=\"1.0\" encoding=\"UTF-8\"?&gt;\n"
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/Constatnts.java]: rename to OHS

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+import javax.xml.soap.SOAPPart;
+
+public class MessageSender {
+
+  public int sendResponseMessage(String data) {
+
+    return 1;
+  }
+
+  private static void callSoapWebService(String soapEndpointUrl, String soapAction) {
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java]: Is this used anywhere?

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+      // Print the SOAP Response
+      System.out.println("Response SOAP Message:");
+      soapResponse.writeTo(System.out);
+      System.out.println();
+
+      soapConnection.close();
+    } catch (Exception e) {
+      System.err.println("\nError occurred while sending SOAP Request to Server!\nMake sure you have the correct endpoint URL and SOAPAction!\n");
+      e.printStackTrace();
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java]: throw an exception so the test will fail

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+      SOAPMessage soapResponse = soapConnection.call(createSOAPRequest(soapAction), soapEndpointUrl);
+
+      // Print the SOAP Response
+      System.out.println("Response SOAP Message:");
+      soapResponse.writeTo(System.out);
+      System.out.println();
+
+      soapConnection.close();
+    } catch (Exception e) {
+      System.err.println("\nError occurred while sending SOAP Request to Server!\nMake sure you have the correct endpoint URL and SOAPAction!\n");
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java]: log.error

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+
+    createSoapEnvelope(soapMessage);
+
+    MimeHeaders headers = soapMessage.getMimeHeaders();
+    headers.addHeader("SOAPAction", soapAction);
+
+    soapMessage.saveChanges();
+
+    /* Print the request message, just for debugging purposes */
+    System.out.println("Request SOAP Message:");
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java]: logger

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+    soapMessage.writeTo(System.out);
+    System.out.println("\n");
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java]: logger

**Review by sdiaz1000 — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/Constatnts.java (hunk 10 above)
Comment (JacobHarrisonQA) [reply src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/Constatnts.java]: Kierans code

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java (hunk 11 above)
Comment (JacobHarrisonQA) [reply src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java]: Kierans code

File: src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java (hunk 13 above)
Comment (JacobHarrisonQA) [reply src/test/java/uk/gov/ons/fwmt/acceptancetests/utils/MessageSender.java]: Kierans code

**Pull request merged**